.github/actions/setup-node/action.yml

```diff
@@ -18,13 +23,43 @@ runs:
       with:
         path: |
           ~/.cargo
-        key: ${{ hashFiles('.github/workflows/cache.yml') }}-${{ runner.os }}-${{ env.OS_ARCH }}
+        key: home-cargo-${{ hashFiles('.github/**/*.yml') }}
+      if: ${{ inputs.read-only-cache != 'true' }}
     - uses: actions/cache@v3
       # Cache first-party code dependencies
       with:
         path: |
           .cargo
-        key: ${{ hashFiles('.github/workflows/cache.yml') }}-${{ runner.os }}-${{ env.OS_ARCH }}-${{ hashFiles('sources/Cargo.lock') }}
+        key: cargo-${{ hashFiles('.github/**/*.yml') }}${{ hashFiles('sources/Cargo.lock') }}
+      if: ${{ inputs.read-only-cache != 'true' }}
+    - uses: actions/cache@v3
+      # Cache build artifacts in 'variants/target/' to reduce rebuilds
+      with:
+        path: |
+          variants/target
+        key: target-${{ hashFiles('.github/**/*.yml') }}${{ hashFiles('packages/**/Cargo.toml', 'packages/**/*.spec') }}
+      if: ${{ inputs.read-only-cache != 'true' }}
+    - uses: actions/cache/restore@v3
+      # Cache `cargo-make`, `cargo-cache`, `cargo-sweep`
+      with:
+        path: |
+          ~/.cargo
+        key: home-cargo-${{ hashFiles('.github/**/*.yml') }}
+      if: ${{ inputs.read-only-cache == 'true' }}
+    - uses: actions/cache/restore@v3
+      # Cache first-party code dependencies
+      with:
+        path: |
+          .cargo
+        key: cargo-${{ hashFiles('.github/**/*.yml') }}${{ hashFiles('sources/Cargo.lock') }}
+      if: ${{ inputs.read-only-cache == 'true' }}
+    - uses: actions/cache/restore@v3
+      # Cache build artifacts in 'variants/target/' to reduce rebuilds
+      with:
+        path: |
+          variants/target
+        key: target-${{ hashFiles('.github/**/*.yml') }}${{ hashFiles('packages/**/Cargo.toml', 'packages/**/*.spec') }}
```

Comment:
If we retain the `cargo` metadata under `variants/target`, but lose the RPMs under `build/rpms` because they're evicted from the cache for size reasons, then the builds will break.

Reply:
Cache eviction works by evicting the older cache items first, so it would need to be a pretty severe cache issue to lost part of the same build artifacts.